app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context.rb

```diff
@@ -0,0 +1,190 @@
+      case @target
+      when ContainerNode then collect_node_metrics
+      when Container then collect_container_metrics
+      when ContainerGroup then collect_group_metrics
+      else raise TargetValidationError, "Validation error: unknown target"
+      end
+    end
+
+    private
+
+    CPU_NANOSECONDS = 1e09
+
+    def target_name
+      "#{@target.class.name.demodulize}(#{@target.id})"
+    end
+
+    def validate_target
+      unless @ext_management_system
+        raise TargetValidationError, "Validation error: ems not defined"
+      end
+      unless @node_cores.to_i > 0
+        raise TargetValidationError, "Validation error: cores not defined"
+      end
+      unless @node_memory.to_i > 0
+        raise TargetValidationError, "Validation error: memory not defined"
+      end
+    end
+
+    def collect_node_metrics
+      cpu_resid = "machine/#{@target.name}/cpu/usage"
+      process_cpu_counters_rate(fetch_counters_rate(cpu_resid))
+
+      mem_resid = "machine/#{@target.name}/memory/usage"
+      process_mem_gauges_data(fetch_gauges_data(mem_resid))
+    end
+
+    def collect_container_metrics
+      group_id = @target.container_group.ems_ref
+
+      cpu_resid = "#{@target.name}/#{group_id}/cpu/usage"
+      process_cpu_counters_rate(fetch_counters_rate(cpu_resid))
+
+      mem_resid = "#{@target.name}/#{group_id}/memory/usage"
+      process_mem_gauges_data(fetch_gauges_data(mem_resid))
+    end
+
+    def collect_group_metrics
+      group_id = @target.ems_ref
+
+      cpu_counters = @target.containers.collect do |c|
+        fetch_counters_rate("#{c.name}/#{group_id}/cpu/usage")
+      end
+      process_cpu_counters_rate(compute_summation(cpu_counters))
+
+      mem_gauges = @target.containers.collect do |c|
+        fetch_gauges_data("#{c.name}/#{group_id}/memory/usage")
+      end
+      process_mem_gauges_data(compute_summation(mem_gauges))
+    end
+
+    def hawkular_client
+      @client ||= Hawkular::Metrics::Client.new(
+        hakwular_entrypoint, hakwular_credentials, hakwular_options)
+    end
+
+    def hakwular_entrypoint
+      worker_class = ManageIQ::Providers::Kubernetes:: \
+                     ContainerManager::MetricsCollectorWorker
+      URI::HTTPS.build(
+        :host => @ext_management_system.hostname,
+        :port => worker_class.worker_settings[:hawkular_port],
+        :path => worker_class.worker_settings[:hawkular_path])
+    end
+
+    def hakwular_credentials
+      {:token => @ext_management_system.try(:authentication_token)}
+    end
+
+    def hakwular_options
+      {:tenant     => @tenant,
+       :verify_ssl => @ext_management_system.verify_ssl_mode}
+    end
+
+    def fetch_counters_rate(resource)
+      compute_derivative(fetch_counters_data(resource))
+    end
+
+    def fetch_counters_data(resource)
+      sort_and_normalize(
+        hawkular_client.counters.get_data(
+          resource,
+          :starts         => (@start_time.to_i - @interval) * 1000,
```

Comment:
Can you put a minor comment here at the end, like `# ms`, to show this is taking milliseconds...It's just easier to a casual reader to see ms and know what it's supposed to take than to have to do the math in their heads.